tests/sentry/tasks/test_derive_code_mappings.py

```diff
@@ -294,6 +294,20 @@ def test_derive_code_mappings_starts_with_app(self):
             assert code_mapping.source_root == ""
             assert code_mapping.repository.name == repo_name
 
+    def test_derive_code_mappings_starts_with_app_complex(self):
+        repo_name = "foo/bar"
+        with patch(
+            "sentry.integrations.github.client.GitHubClientMixin.get_trees_for_org"
+        ) as mock_get_trees_for_org:
+            mock_get_trees_for_org.return_value = {
+                repo_name: RepoTree(Repo(repo_name, "master"), ["sentry/utils/errors.js"])
+            }
+            derive_code_mappings(self.project.id, self.event_data)
+            code_mapping = RepositoryProjectPathConfig.objects.all()[0]
+            assert code_mapping.stack_root == "app:///"
+            assert code_mapping.source_root == "sentry/"
+            assert code_mapping.repository.name == repo_name
+
```

Comment:
This was a test case we did not cover for before. The logic in the reverted PR would return "app://" instead of "app:///".